pkg/agent/cniserver/pod_monitor.go

```
@@ -0,0 +1,243 @@
+		ovsPort := string(bytes.Trim(status.Desc.Name, "\x00"))
+		ofPort := status.Desc.PortNo
+		m.unreadyMapMutex.Lock()
+		defer m.unreadyMapMutex.Unlock()
+		podInfo, found := m.unreadyInterfaces[ovsPort]
+		if !found {
+			return
+		}
+		podInfo.ofPort = int32(ofPort)
+		m.queue.Add(ovsPort)
+	}
+}
+
+func (m *podIfaceMonitor) processNextWorkItem() bool {
+	key, quit := m.queue.Get()
+	if quit {
+		return false
+	}
+	defer m.queue.Done(key)
+
+	if err := m.updateUnreadyPod(key); err != nil {
+		// Put the item back on the workqueue to handle any transient errors.
+		m.queue.AddAfter(key, retryInterval)
+	}
+	return true
+}
+
+// worker is a long-running function that will continually call the processNextWorkItem function in
+// order to read and process a message on the workqueue.
+func (m *podIfaceMonitor) worker() {
+	for m.processNextWorkItem() {
+	}
+}
+
+func (m *podIfaceMonitor) updateUnreadyPod(ovsPort string) error {
+	m.unreadyMapMutex.Lock()
+	defer m.unreadyMapMutex.Unlock()
+	podInfo, found := m.unreadyInterfaces[ovsPort]
+	if !found {
+		klog.InfoS("OVS port is not found", "ovsPort", ovsPort)
+		return nil
+	}
+
+	ifConfig, found := m.ifaceStore.GetInterfaceByName(ovsPort)
+	if !found {
+		klog.InfoS("Interface config is not found in local cache, remove from local cache", "port", ovsPort)
+		delete(m.unreadyInterfaces, ovsPort)
+		return nil
+	}
+
+	// Add Pod ready event if the pod flows are successfully installed. This is supposed to be a retry case that
+	// Pod ready event fails to add in the last round.
+	if podInfo.flowsInstalled {
+		if err := m.processPodEvents(podInfo); err != nil {
+			return err
+		}
+		// Remove un-ready pod from local cache.
+		delete(m.unreadyInterfaces, ovsPort)
+		return nil
+	}
+
+	if podInfo.ofPort == 0 {
+		// Add Pod not-ready event if the pod flows are not successfully installed, and the OpenFlow port is not allocated.
+		// Returns error so that we can have a retry after 5s.
+		_ = m.processPodEvents(podInfo)
```

Is there any risk to ignore the error returned by processPodEvents?